frontend/src/App.tsx

```diff
@@ -1441,6 +1457,12 @@ export class App extends PureComponent<Props, State> {
         })
       : null
 
+    const showDeployButton =
+      isLocalhost() &&
```

Comment:
In PR: https://github.com/streamlit/streamlit/pull/6174 I want to limit the use of the `isLocalhost` function, because some people want to have this behavior also on a remote server, e.g. Docker.

Reply:
One of us has to merge first and second one adjust, for now whoever gets first 👍